sieve_of_eratosthenes/sieveOfEratosthenes.c

```diff
@@ -0,0 +1,34 @@
+#include <stdio.h>
+#include <string.h>
+#include <math.h>     // for using square root 
+#include <stdbool.h>  // to include bool datatype 
+
+void sieveOfEratosthenes(int n) {
+	bool primes[n+1];
+	int i, sqrtOfn = sqrt(n)+1, j;
+	
+	memset(primes, true, sizeof(primes));
+	primes[0] = false;
+	primes[1] = false;
+
+	for (i = 2; i < sqrtOfn; i++) {
+		if (primes[i]) {
+			for (j = 2*i; j < n+1; j += i) {
+				primes[j] = false;
+			}
+		}
+	}
+
+	for (j = 2; j < n; j++) {
+		if (primes[j]) {
+			printf("%d ", j);
```

Comment:
That doesn't works because you give out a completely new memory location to `primes` in that function. One way to get things going would be -
```c
void sieveOfEratosthenes(int n, bool *primes) {
	...
	memset(primes, true, n+1);
	...
}

int main() {
        ...
	bool primes[n+1];
	sieveOfEratosthenes(n, primes);
	...
}
```

But you can buff up your function -
```c
bool *sieveOfEratosthenes(int n) {
	...
        primes = (bool*)calloc(n+1, sizeof(bool));
	memset(primes, true, n+1);
	...
        return primes;
}

int main() {
        ...
	bool *primes;
	primes = sieveOfEratosthenes(n, primes);
	...
}
```